docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,562 @@
+```
+
+As you can see, the `User` with id 1 was fetched only once in a single round-trip. The next
+time it was needed we used the cached versions, thus avoiding another request to the user data
+source.
+
+## Combining data from multiple sources
+
+Now that we know about some of the optimizations that Fetch can make to read data efficiently,
+let's look at how we can combine more than one data source. Imagine that we are rendering a blog
+and have the following types for posts and post information:
+
+```scala
+type PostId = Int
+case class Post(id: PostId, author: UserId, content: String)
+case class PostInfo(topic: String)
+```
+
+As you can see, every `Post` has an author, but it refers to it by its id. We'll implement two data sources:
+
+- one for retrieving a post given a post id
+- another for retrieveng post metadata given a post id
+
+```scala
+implicit object PostSource extends DataSource[PostId, Post]{
+  override def fetch(ids: List[PostId]): Eval[Map[PostId, Post]] = {
+    Eval.later({
+      println(s"Fetching posts $ids")
+      ids.map(i => (i, Post(i, i % 3 + 1, s"An article with id $i"))).toMap
+    })
+  }
+}
+
+def getPost(id: PostId): Fetch[Post] = Fetch(id)
+
+implicit object PostInfoSource extends DataSource[PostId, PostInfo]{
+  override def fetch(ids: List[PostId]): Eval[Map[PostId, PostInfo]] = {
+    Eval.later({
+      println(s"Fetching post info $ids")
+      ids.map(i => (i, PostInfo(if (i % 2 == 0) "applicative" else "monad"))).toMap
+    })
+  }
+}
+
+def getPostInfo(id: PostId): Fetch[PostInfo] = Fetch(id)
+```
+
+We can also implement a function for fetching a post's author given a post:
+
+```scala
+def getAuthor(p: Post): Fetch[User] = Fetch(p.author)
+```
+
+Now that we have multiple sources, let's mix them in the same fetch.
+
+```scala
+val fch: Fetch[(Post, User)] = for {
+  post <- getPost(1)
+  user <- getAuthor(post)
+} yield (post, user)
+
+val result: (Post, User) = Fetch.run[Id](fch)
+// Fetching posts List(1)
+// Fetching users List(2)
+
+//=> result: (Post, User) = (Post(1,2,An article with id 1),User(2,@egg_2))
+```
+
+In the previous example we fetched a post given its id, and then fetched its author. These
+data could come from entirely different places but Fetch makes working with heterogeneous sources
+of data very easy.
+
+### Parallelism
+
+Combining multiple independent requests to the same data source results in batching but what happens
+when combining independent requests to *different* data sources? We have the oportunity to query the
+different data sources at the same time.
+
+```scala
+val fch: Fetch[(Post, User)] = getPost(1).product(getUser(2))
+
+val result: (Post, User) = Fetch.run[Id](fch)
+// Fetching posts List(1)
+// Fetching users List(2)
+
+//=> result: (Post, User) = (Post(1,2,An article with id 1),User(2,@egg_2))
+```
+
+Since we are interpreting the fetch to the `Id` monad, that doesn't give us any concurrency, the fetches
```

Comment:
We may want to describe here or somewhere more appropiate that Fetches are reusable and can be interpreted to different runtimes for example Task or Future in production code and Id in tests